File: src/main/java/com/example/AI_Study_Planer/dto/response/Quiz/QuizDashboardResponse.java
package com.example.AI_Study_Planer.dto.response.Quiz;

import lombok.Data;

import java.util.List;

@Data
public class QuizDashboardResponse {

    private List<QuizCardResponse> recommendedQuizzes;

    private DailyQuizResponse dailyQuiz;

    private List<QuizCardResponse> continueQuizzes;

    private List<WeakTopicResponse> weakTopics;

    private List<QuizHistoryResponse> recentQuizzes;

    private QuizStatsResponse stats;

    private String aiTip;
}

File: src/types/Response/Quiz/QuizDashboard/QuizDashboardResponse.ts
import type { QuizCardResponse } from "@/types/Response";
import type { WeakTopicResponse } from "./WeakTopicResponse";
import type { QuizHistoryResponse } from "./QuizHistoryResponse";
import type { QuizStatsResponse } from "./QuizStatsResponse";
import type { DailyQuizResponse } from "./DailyQuizResponse";

export interface QuizDashboardResponse {
  recommendedQuizzes: QuizCardResponse[];
  dailyQuiz: DailyQuizResponse;
  continuedQuizzes: QuizCardResponse[];
  weakTopics: WeakTopicResponse[];
  recentQuizzes: QuizHistoryResponse[];
  stats: QuizStatsResponse;
  aiTips: string[];
}
